compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -221,6 +225,84 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    fn compute_2229_migrations(&self, closure_def_id: DefId, closure_span: Span) {
+        let upvars = if let Some(upvars) = self.tcx.upvars_mentioned(closure_def_id) {
+            upvars
+        } else {
+            return;
+        };
+
+        let mut need_migrations = Vec::new();
+
+        let typeck_results = self.typeck_results.borrow();
+        let min_captures = typeck_results.closure_min_captures.get(&closure_def_id);
+
+        for (&var_hir_id, _) in upvars.iter() {
+            let ty = self.node_ty(var_hir_id);
+            let ty = self.infcx.shallow_resolve(ty);
+            if ty.is_ty_infer() {
```

Comment:
If I don't shallow resolve this gets triggered: http://csclub.uwaterloo.ca/~a52arora/rust-builds/log_a162b52d-0e8e-4d3a-b4fe-9cfbce7b1780